functions/Get-DbaDbFileMap.ps1

```diff
@@ -0,0 +1,90 @@
+function Get-DbaDbFileMap {
+    <#
+    .SYNOPSIS
+        Provides an easy way to build the FileMapping Hashtable for Restore-DbaDatabase from an existing database
+
+    .DESCRIPTION
+        Provides an easy way to build the FileMapping Hashtable for Restore-DbaDatabase from an existing database
+
+    .PARAMETER SqlInstance
+        The target SQL Server instance or instances. This can be a collection and receive pipeline input.
+
+    .PARAMETER SqlCredential
+        Login to the target instance using alternative credentials. Accepts PowerShell credentials (Get-Credential).
+
+        Windows Authentication, SQL Server Authentication, Active Directory - Password, and Active Directory - Integrated are all supported.
+
+        For MFA support, please use Connect-DbaInstance.
+
+    .PARAMETER Database
+        The database(s) to process - this list is auto-populated from the server. If unspecified, all databases will be processed.
+
+   .PARAMETER InputObject
+        Database object piped in from Get-DbaDatabase
+
+    .PARAMETER EnableException
+        By default, when something goes wrong we try to catch it, interpret it and give you a friendly warning message.
+        This avoids overwhelming you with "sea of red" exceptions, but is inconvenient because it basically disables advanced scripting.
+        Using this switch turns this "nice by default" feature off and enables you to catch exceptions with your own try/catch.
+
+    .NOTES
+        Tags: Database
+        Author: Chrissy LeMaire (@cl), netnerds.net | Andreas Jordan (@JordanOrdix), ordix.de
+
+        Website: https://dbatools.io
+        Copyright: (c) 2021 by dbatools, licensed under MIT
+        License: MIT https://opensource.org/licenses/MIT
+
+    .LINK
+        https://dbatools.io/Get-DbaDbFileMap
+
+    .EXAMPLE
+        PS C:\> $filemap = Get-DbaDbFileMap -SqlInstance sql2016 -Database test
+        PS C:\> Get-ChildItem \\nas\db\backups\test | Restore-DbaDatabase -SqlInstance sql2019 -Database test -FileMapping $filemap.FileMap
+
+        Restores test to sql2019 using the file structure built from the existing database on sql2016
+    #>
+    [CmdletBinding()]
+    param ([parameter(ValueFromPipeline)]
+        [DbaInstanceParameter[]]$SqlInstance,
+        [PSCredential]$SqlCredential,
+        [object[]]$Database,
```

Comment:
I would use [string[]] here to indicate that it's the database name and not a database SMO object.

Reply:
We use `[object[]]` @andreasjordan because TEPP populates the list of Databases for dynamic validate sets. If you use `string` type this won't be accessible.

Reply:
Gives this ability when you use `[object[]` on a parameter where we have the backend setup for TEPP.

![image](https://user-images.githubusercontent.com/11204251/126071259-abece01f-7d4d-49aa-9520-024edee54b10.png)

Reply:
Ok, didn't know that. Thanks. Will have a look at the commands I changed in the past.

Reply:
We can add pretty much anything we can get from SMO or t-sql to list out. Primary one is database. Not the easiest system to deal with and not always consistent in some environments I touch.

Reply:
wait for real? i had used object in the past to allow both SMO and text but then went to string[] instead after learning to use `InputObject` better. I'll see if string[] destroys this

Reply:
It does 😉 because TEPP has to be an object type to return the list to tab completor.

Reply:
We covered a few initially and at one point I had added in some as I worked on commands, but in that phase of maintainers taking a vacation for a bit 🍹 I never got back to adding support for the main parameters in commands.

Reply:
yay, just tested, `[string[]]` does work 💯  updated